fixes #250 and the broken links in the readme

chinito opened the pull request:
This fixes broken links in the readme

ptrimble commented on CONTRIBUTING.md:
```diff
@@ -15,12 +15,12 @@ or, if you forgot to do so, run `make update_defs` so that all of the submodule
 Our definitions are written in YAML. They are housed in a [separate repository](https://github.com/holidays/definitions) so
 that they can be used by tools written in other languages. You can find a complete guide to our format in the
-[definitions SYNTAX guide](https://github.com/holidays/definitions/SYNTAX.md).
+[definitions SYNTAX guide](https://github.com/holidays/definitions/blob/master/SYNTAX.md).
```
Oops! Thank you for the fix!

ptrimble commented on lib/holidays/finder/rules/in_region.rb:
```diff
@@ -9,7 +9,15 @@ def call(requested, available)
+              if r.to_s =~ /_/
+                chunks = r.to_s.split('_')
+                
+                chunks.length.downto(1).map do |num|
```
I think this is a great change! What do you think about adding a test in the specific `test_in_region.rb` file to show the failure specifically so we can make sure we don't introduce a similar bug in the future?

chinito replied:
sure, will do as soon as I got time. Right now this is only testet with the definition. Weekend is near, so I hope to bring it in then.

ptrimble commented on test/defs/test_defs_de.rb:
```diff
@@ -23,11 +23,11 @@ def test_de
 end
 
-[:de_bw, :de_by, :de_st, :de_].each do |r|
+[:de_bw, :de_by, :de_by_aux, :de_st, :de_].each do |r|
```
Was this added because of the changes in https://github.com/holidays/definitions/pull/16? If so, you can remove them from this PR, I will handle re-generating the files for you. You don't have to worry about it! I would prefer if the PR just concentrated on the README and the actual bug in question, I'll handle the definitions. 👍

chinito replied:
I added this so it tests the holidays of bavaria also valid in augsburg. It doesn't really have todo with holidays/definitions#16

chinito commented:
With the submodule you're totally right. I removed it, because travis was failing. Maybe you can help me understand how this works?

At first I cloned definitions on their own and did the changes from holidays/definitions#16 then I saw the problem is not in definitions but in the gem.

So I headed there and cloned this. After some experiments I found this make file with it's make generate. So I did that, than got in the cloned definitions folder. I then made the test for `:de_by_aux`to find out where the problem is.
After I fixed it and the tests were all green. I committed the my changes to the gem and wanted to push to the repo, but now I head the definitions in the tree and this was not possible.
So I got to the changed definitions and commited there. Afterwards I could push. Then I did the Pull Request and travis failed with a problem on the definitions. You can see that in the history, it's the fix after the README. So I tried to find the problem but, the only solution was to remove them. Then Travis run again. Now the tests worked.

So I came up with the following solution to make that work for me. Change definition submodule in the holiday gem working dir. Then run make generate. This changed the holiday gem generated tests. When applying fix and new tests run, remove definitions and send pull request.

So it boils down to, I could not send pull request as long as I had changed definitions in the working tree. Maybe you can add something to the readme how to handle this?

chinito commented:
So I hope the test is ok, like I did it. I'm not so used to Testunit and work normally with rspec. If you have any suggestions, please let me know.

ptrimble commented:
Yep! Looks good to me. I will handle getting a release out sometime this week (or maybe next week depending on when my baby is born 😄 )

ttwo32 commented:
:+1:

chinito commented:
👍
